File: vl10/rational.cpp
#include <iostream>
#include "rational.h"

rational::rational(int zaehler, int nenner): zaehler{zaehler},nenner{nenner}{}
/*
rational rational::operator*(const rational& r){
    return rational{this->zaehler*r.zaehler,this->nenner*r.nenner};
}
*/

rational::rational(int zaehler): rational{zaehler,1} {}

rational operator*(const rational& l, const rational& r){
   /* rational out;
    out.zaehler = l.zaehler * r.zaehler;
    out.nenner = l.nenner* r.nenner;
    return out;*/
    return rational{l.zaehler*r.zaehler,l.nenner*r.nenner};
}

rational rational::operator*(const int k) const{ //rational * k
    return rational{this->zaehler * k,this->nenner};
}

rational operator*(const int k,const rational& r){
    //return rational{r.zaehler * k, r.nenner};
    return r*k;
}

std::ostream& operator<<(std::ostream& out,const rational& r){
    out << "(" << r.zaehler << "/" << r.nenner << ")";
    return out;
}

File: Wiederholung/praktikum13/rational.cs
class Rational
{
    public int Zaehler { get; set; }
    public int Nenner { get; set; }

    public Rational(int zaehler, int nenner)
    {
        Zaehler = zaehler;
        Nenner = nenner;
    }

    public override string ToString()
    {
        return string.Format("{0}/{1}", Zaehler, Nenner);
    }
    public static Rational operator *(Rational l, Rational r)
    {
        if (l.Nenner != 0 && r.Nenner != 0)
        {
            Rational neu;
            neu.Zaehler = l.Zaehler * r.Zaehler;
            neu.Nenner = l.Nenner * r * Nenner;
            return neu;
        }
        else
        {
            throw new Exception("Error: Nenner ist null");
        }
    }
}